### PR TITLE
Split workflow deletion to two steps. delete current exe and workflow exe

### PR DESCRIPTION
**Pull request by yux0**
*Split workflow deletion to two steps. delete current exe and workflow exe
Cassandra: 
1. updateWorkflowExecution update current exe
2. Use timer to delete current exe and workflow exe

MySQL:
1. Split the delete current exe and workflow exe to two APIS

#1772

### CODE REVIEW

File: common/metrics/defs.go [hunk 1]
```diff
@@ -142,6 +142,8 @@ const (
 	PersistenceResetWorkflowExecutionScope
 	// PersistenceDeleteWorkflowExecutionScope tracks DeleteWorkflowExecution calls made by service to persistence layer
 	PersistenceDeleteWorkflowExecutionScope
+	// PersistenceDeleteWorkflowCurrentExecutionScope tracks DeleteWorkflowCurrentExecution calls made by service to persistence layer
+	PersistenceDeleteWorkflowCurrentExecutionScope
```
Comment (wxing1292) [inline common/metrics/defs.go]: DeleteCurrentWorkflowExecution

File: common/mocks/ExecutionManager.go [hunk 2]
```diff
@@ -167,6 +167,20 @@ func (_m *ExecutionManager) DeleteWorkflowExecution(request *persistence.DeleteW
 	return r0
 }
 
+// DeleteWorkflowCurrentExecution provides a mock function with given fields: request
+func (_m *ExecutionManager) DeleteWorkflowCurrentExecution(request *persistence.DeleteWorkflowExecutionRequest) error {
```
Comment (wxing1292) [inline common/mocks/ExecutionManager.go]: DeleteCurrentWorkflowExecution

File: common/persistence/cassandra/cassandraPersistence.go [hunk 3]
```diff
@@ -2289,7 +2261,7 @@ func (d *cassandraPersistence) DeleteWorkflowExecution(request *p.DeleteWorkflow
 	return nil
 }
 
-func (d *cassandraPersistence) DeleteWorkflowCurrentRow(request *p.DeleteWorkflowExecutionRequest) error {
+func (d *cassandraPersistence) DeleteWorkflowCurrentExecution(request *p.DeleteWorkflowExecutionRequest) error {
```
Comment (wxing1292) [inline common/persistence/cassandra/cassandraPersistence.go]: DeleteCurrentWorkflowExecution

File: common/persistence/dataInterfaces.go [hunk 4]
```diff
@@ -1381,6 +1379,7 @@ type (
 		ResetMutableState(request *ResetMutableStateRequest) error
 		ResetWorkflowExecution(request *ResetWorkflowExecutionRequest) error
 		DeleteWorkflowExecution(request *DeleteWorkflowExecutionRequest) error
+		DeleteWorkflowCurrentExecution(request *DeleteWorkflowExecutionRequest) error
```
Comment (wxing1292) [inline common/persistence/dataInterfaces.go]: DeleteCurrentWorkflowExecution

Comment (wxing1292) [reply common/persistence/dataInterfaces.go]: DeleteCurrentWorkflowExecutionRequest

File: common/persistence/executionStore.go [hunk 5]
```diff
@@ -689,6 +687,10 @@ func (m *executionManagerImpl) DeleteWorkflowExecution(request *DeleteWorkflowEx
 	return m.persistence.DeleteWorkflowExecution(request)
 }
 
+func (m *executionManagerImpl) DeleteWorkflowCurrentExecution(request *DeleteWorkflowExecutionRequest) error {
```
Comment (wxing1292) [inline common/persistence/executionStore.go]: DeleteCurrentWorkflowExecution

File: common/persistence/persistence-tests/executionManagerTest.go [hunk 6]
```diff
@@ -920,14 +917,28 @@ func (s *ExecutionManagerSuite) TestDeleteCurrentWorkflow() {
...
+		WorkflowID: info0.ExecutionInfo.WorkflowID,
+		RunID:      uuid.New(),
+	}
+
+	// test wrong run id with conditional delete
+	s.DeleteWorkflowCurrentExecution(fakeInfo)
+
+	runID5, err5 := s.GetCurrentWorkflowRunID(domainID, *workflowExecution.WorkflowId)
+	s.NoError(err5)
+	s.Equal(*workflowExecution.RunId, runID5)
```
Comment (wxing1292) [inline common/persistence/persistence-tests/executionManagerTest.go]: GetRunId()

File: common/persistence/persistence-tests/persistenceTestBase.go [hunk 7]
```diff
@@ -902,6 +900,15 @@ func (s *TestBase) DeleteWorkflowExecution(info *p.WorkflowExecutionInfo) error
 	})
 }
 
+// DeleteWorkflowCurrentExecution is a utility method to delete the workflow current execution
+func (s *TestBase) DeleteWorkflowCurrentExecution(info *p.WorkflowExecutionInfo) error {
```
Comment (wxing1292) [inline common/persistence/persistence-tests/persistenceTestBase.go]: DeleteCurrentWorkflowExecution

File: common/persistence/persistenceInterface.go [hunk 8]
```diff
@@ -66,6 +66,7 @@ type (
 
 		CreateWorkflowExecution(request *InternalCreateWorkflowExecutionRequest) (*CreateWorkflowExecutionResponse, error)
 		DeleteWorkflowExecution(request *DeleteWorkflowExecutionRequest) error
+		DeleteWorkflowCurrentExecution(request *DeleteWorkflowExecutionRequest) error
```
Comment (wxing1292) [inline common/persistence/persistenceInterface.go]: DeleteCurrentWorkflowExecution

File: common/metrics/defs.go (hunk 1 above)
Comment (yux0) [reply common/metrics/defs.go]: Done

File: common/mocks/ExecutionManager.go (hunk 2 above)
Comment (yux0) [reply common/mocks/ExecutionManager.go]: Done

File: common/persistence/cassandra/cassandraPersistence.go (hunk 3 above)
Comment (yux0) [reply common/persistence/cassandra/cassandraPersistence.go]: Done

File: common/persistence/executionStore.go (hunk 5 above)
Comment (yux0) [reply common/persistence/executionStore.go]: Done

File: common/persistence/persistence-tests/executionManagerTest.go (hunk 6 above)
Comment (yux0) [reply common/persistence/persistence-tests/executionManagerTest.go]: Done

File: common/persistence/persistence-tests/persistenceTestBase.go (hunk 7 above)
Comment (yux0) [reply common/persistence/persistence-tests/persistenceTestBase.go]: Done

File: common/persistence/persistenceInterface.go (hunk 8 above)
Comment (yux0) [reply common/persistence/persistenceInterface.go]: Done

File: common/persistence/dataInterfaces.go (hunk 4 above)
Comment (yux0) [reply common/persistence/dataInterfaces.go]: Done

**Pull request merged**